pkg/icingaredis/client.go

```diff
@@ -89,94 +87,78 @@ func (c *Client) HYield(ctx context.Context, key string) (<-chan HPair, <-chan e
 				return WrapCmdErr(cmd)
 			}
 
-			g.Go(func(page []string) func() error {
-				return func() error {
-					for i := 0; i < len(page); i += 2 {
-						select {
-						case pairs <- HPair{
-							Field: page[i],
-							Value: page[i+1],
-						}:
-							cnt.Inc()
-						case <-ctx.Done():
-							return ctx.Err()
-						}
-					}
-
-					return nil
+			for i := 0; i < len(page); i += 2 {
+				select {
+				case pairs <- HPair{
+					Field: page[i],
+					Value: page[i+1],
+				}:
+					cnt.Inc()
+				case <-ctx.Done():
+					return ctx.Err()
 				}
-			}(page))
+			}
 
 			if cursor == 0 {
 				break
 			}
 		}
 
-		return g.Wait()
-	})
-
-	return pairs, com.WaitAsync(g)
+		return nil
+	}))
 }
 
 // HMYield yields HPair field-value pairs for the specified fields in the hash stored at key.
 func (c *Client) HMYield(ctx context.Context, key string, fields ...string) (<-chan HPair, <-chan error) {
 	pairs := make(chan HPair)
```

Comment:
(This on is probably fine to keep unbuffered as there's some implicit buffering by using multiple goroutines.)